packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -134,11 +135,23 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
     let { formatDate, isRequired, strings, value } = nextProps;
     const errorMessage = (isRequired && !value) ? (strings.isRequiredErrorMessage || '*') : null;
 
-    this.setState({
-      selectedDate: value || new Date(),
-      formattedDate: (formatDate && value) ? formatDate(value) : '',
-      errorMessage: errorMessage
-    });
+    if (errorMessage != null) {
+      this.setState({
+        errorMessage: errorMessage
+      });
+    }
+    else {
+      // Check if the date value changed from old props value, i.e., if indeed a new date is being
+      // passed in and only then update selectedDate
+      let oldValue = this.props.value;
+      if (!compareDates(oldValue, value)) {
+        this.setState({
+          selectedDate: value || new Date(),
+          formattedDate: (formatDate && value) ? formatDate(value) : '',
```

Comment:
From what I read in the docs, setting state in componentWillUpdate should be avoided, @MLoughry, did you mean something else? We could use the setState callback in the onSelectDate method and this works by basically calling setState again in the callback for setState, but this is kind of hacky and I guess we could have a valid case where the user sent us a new date value as the result of the callback that we would now be ignoring if we force set the date this way. Probably not a common scenario but... One thing we can do here is change the code so that we still have the bug in case the user passed in a new format. I don't foresee this will be a common scenario so hopefully this would not be a very common use case. Thoughts?